### ISSUE TITLE
need help building the sub modules

### ISSUE THREAD

**Issue by pravi** [lodash/lodash#2868]
We are trying to build sub modules as part of debian package creation, but we are not able to get it working.
More details here https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=845227 can someone help?

**Comment by jdalton** [comment]
Hi @pravi

lodash.assignwith Is it's [own package](https://www.npmjs.com/package/lodash.assignwith)

**Comment by pravi** [comment]
@jdalton are you saying lodash.assignwith is also developed in a separate repository? Its package page is giving https://github.com/lodash/lodash as the repository. I can't find it there. I assume they are generated, I need help generating lodash.assignwith from sources in https://github.com/lodash/lodash

**Comment by jdalton** [comment]
You can find the sources for it in the `npm-packages` branch [here](https://github.com/lodash/lodash/tree/npm-packages/lodash.assignwith) and by tag like [this](https://github.com/lodash/lodash/tree/4.2.0-npm-packages/lodash.assignwith).

**Comment by jdalton** [comment]
You shouldn't need to rebuild from source though. Npm offers the .tgz files of each release.
Use `npm info lodash`:
```bash
{ shasum: '34a3055babe04ce42467b607d700072c7ff6bf42',
     tarball: 'https://registry.npmjs.org/lodash/-/lodash-4.17.2.tgz' },
  directories: {} }
```

**Comment by pravi** [comment]
@jdalton we prefer to build from the source repo. Can you add tags for the npm-packages branch so I can easily package new versions? Right now I can take a snapshot of that branch.

**Comment by jdalton** [comment]
You shouldn't need to build from the source. It's better to simply use the packages as provided. That way there's less likely to be user error, especially since you're not familiar with the project. Is there someone else I can talk to or can I file a formal complaint.

**Comment by pravi** [comment]
You can respond to https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=845227 there will be a "Reply" link at the top (It will be mailto address for 845227@bugs.debian.org). You can send an email to this address and all javascript packaging team will receive your message.

If we take the github repo snapshots, we can package all modules from a single tarball.

**Comment by jdalton** [comment]
Thanks!

**Comment by jdalton** [comment]
You've [ended up publishing](https://ftp-master.debian.org/new/node-lodash-packages_4.15.0-1.html) all 407 individual packages as a single package.

**Comment by jdalton** [comment]
Also why are you creating duplicate packages? Why not use the npm registry as is?

**Comment by pravi** [comment]
We want to maintain only a single version of any library. So any package depending on any of these individual modules will need to be updated to use the version available in debian anyway. It definitely means more work for debian package maintainers, but it makes it easy for users. Users have to know only about apt and not npm, rubygem, pip, gopkg or package managers of each language. 

We are not interested to convert all npm modules, but only those we need for applications like grunt, gulp, jison etc. We use these to build other libraries used by applications like gitlab, diaspora etc. I started this process for giving apt-get install diaspora. It needed handlebars-assets ruby gem which bundles libjs-handlebars. We want to be able to build libjs-handlebars from node module handlebars. We packaged jison and grunt already. We are in the process of packaging gulp to be able to build babel. So we want only a gloablly installed app that uses globally installed libraries.

I know it is opposite of what npm does and is hard to digest for you. See https://github.com/LeoIannacone/npm2deb/issues/11 for a discussion on why we do it this way. As mentioned above, if I use npm registry, I will have to package each of these modules separately which is more work and uses more metadata. nodejs will find these as separate modules only.
